exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/DrillFilterAggregateTransposeRule.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.planner.logical;
+
+import org.apache.calcite.rel.rules.FilterAggregateTransposeRule;
+
+public class DrillFilterAggregateTransposeRule{
```

Comment:
If the only thing we are doing is creating a FilterAggregateTransposeRule with a custom filter factory, does it make sense to simply create this in planner phase rather than having a custom rule class? It seems like the rule class isn't actually a rule.

Also, can you put a little information next to the rule definition of what happens if we don't use a custom factory? It would seem like the existing transpose rule would have worked.